source/develop/release-management/features/storage/multipath-events.html.md

```diff
@@ -0,0 +1,193 @@
+- The handler subscribes to multipath events during startup
+- The handler updates the data set of the faulty paths when its callback is called:
+    - PATH_FAILED - Add the faulty path to the map according to the device GUID
+    - PATH_REINSTATED - Remove the faulty path to the map according to the device GUID
+- Statistics are provided to engine via 'Host.getStats' verb.
+
+
+### New engine API
+
+New audit logs are added:
+- Faulty multipath paths on host "HOST_NAME" on devices: "GUID", "GUID" ... 
+- No faulty multipath paths on host "HOST_NAME"
+
+## Benefit to oVirt
+
+- The user will be able to detect storage issues faster.
+
+
+## User Experience
+
+New audit logs are added with information on the device path status.
+The logs are per host.
+Since the length of the information available in the audit log is limited,
+the full information of the failing paths will be available in the logs.
+The events are not generated on "real-time", but are based on polling of
+the statistics. Only if a change in the data received is detected, a new
+audit log will be created
+
+If a single LUN goes down on all the hosts, the user will get an alarm
+for each host in the DC.
+
+If a single path goes down on one of the hosts, the user will get an alarm
+only for this host.
+
+
+## Installation/Upgrade
+
+This feature does not affect engine or Vdsm installation or upgrades
+
+
+## User work-flows
+
+- Once a user connects to a new ISCSI server, no events will be created.
+
+- If there is an issue with the connection to the storage server, and the 
+multipath reports an issue with one or more device path, a "Faulty multipath paths on host" will be created
+
+- Once the issue is fixed, and all paths are up a "No faulty multipath paths on host "HOST_NAME"
+event will be created
+
+- If only part of the paths are fixed, a "Faulty multipath paths on host" will be created with the details of the devices.
+
+
+## Dependencies / Related Features
+
+Additional package dependency in Vdsm:
+- pyudev
+
+No related features.
+
+
+### Entity Description
+
+NA
+
+## CRUD
+
+NA
+
+
+## Event Reporting
+
+- Faulty multipath paths on host "HOST_NAME" on devices: "GUID", "GUID" ... 
+- No faulty multipath paths on host "HOST_NAME"
+
+
+## Documentation / External references
+
+This feature is required for resolving
+[Bug 723931](https://bugzilla.redhat.com/723931)
+
+Pyudev - [pyudev documentation](http://pyudev.readthedocs.io/en/latest/guide.html#monitoring-devices)
+
+Uevent [device mapper events kernel documentation](https://www.kernel.org/doc/Documentation/device-mapper/dm-uevent.txt)
+
+## Testing
+
+### Negative flows
+
+- Block network to ISCSI storage server on a specific host
+  - 'Faulty multipath paths on host "HOST_NAME" on devices: "GUID", "GUID" ...' is triggered
+- Unblock access to iscsi server on a specific host
+  - Wait until all paths go up. a 'No faulty multipath paths on host "HOST_NAME"' is triggered
+
```

Comment:
We need more detailed description that can help QE person to test this.

Reply:
I will add more details later